packages/talos_agent/src/agentv2/state_manager.rs

```diff
@@ -61,13 +62,22 @@ impl StateManager {
 
         let it_for_consumer = self.int_type.clone();
         let config_for_consumer = config.clone();
+
+        let consumer_barrier = Arc::new(Notify::new());
+        let consumer_barrier_arc = Arc::clone(&consumer_barrier);
         tokio::spawn(async move {
```

Comment:
Maybe in future, it would be good to move this thread logic as a separate service all together. It listens to the decisions coming in, and passes it to the core logic of  the `state_manager.rs`, which is to receive messages and manage the state.

Reply:
Done